src/basic/memory-util.c

```diff
@@ -39,3 +39,17 @@ bool memeqbyte(uint8_t byte, const void *data, size_t length) {
         /* Now we know first 16 bytes match, memcmp() with self.  */
         return memcmp(data, p + 16, length) == 0;
 }
+
+void *make_swapped(const void *data, size_t size) {
+        assert(data);
+        assert(size != 0);
+
+        void *p = malloc(size);
+        if (p) {
+                uint8_t *p_dst = (uint8_t*) p;
+                uint8_t *p_src = (uint8_t*) data;
```

Comment:
i'd define this as `const uint8_t*` really.

also, the explicit casts should be unnecessary then as void* casts to any pointer type implicitly.